plugins/LayerView/LayerView.py

```diff
@@ -178,103 +78,29 @@ def event(self, event):
-        self._only_show_top_layers = bool(Preferences.getInstance().getValue("view/only_show_top_layers"))
-
-        self._startUpdateTopLayers()
-
-
-class _CreateTopLayersJob(Job):
-    def __init__(self, scene, layer_number, solid_layers):
-        super().__init__()
-
-        self._scene = scene
-        self._layer_number = layer_number
-        self._solid_layers = solid_layers
-        self._cancel = False
-
-    def run(self):
-        layer_data = None
-        for node in DepthFirstIterator(self._scene.getRoot()):
-            layer_data = node.callDecoration("getLayerData")
-            if layer_data:
-                break
-
-        if self._cancel or not layer_data:
-            return
-
-        layer_mesh = MeshBuilder()
-        for i in range(self._solid_layers):
-            layer_number = self._layer_number - i
-            if layer_number < 0:
-                continue
-
-            try:
-                layer = layer_data.getLayer(layer_number).createMesh()
-            except Exception:
-                Logger.logException("w", "An exception occurred while creating layer mesh.")
-                return
-
-            if not layer or layer.getVertices() is None:
-                continue
-
-            layer_mesh.addIndices(layer_mesh.getVertexCount() + layer.getIndices())
-            layer_mesh.addVertices(layer.getVertices())
-
-            # Scale layer color by a brightness factor based on the current layer number
-            # This will result in a range of 0.5 - 1.0 to multiply colors by.
-            brightness = numpy.ones((1, 4), dtype=numpy.float32) * (2.0 - (i / self._solid_layers)) / 2.0
-            brightness[0, 3] = 1.0
-            layer_mesh.addColors(layer.getColors() * brightness)
-
-            if self._cancel:
-                return
-
-            Job.yieldThread()
+        if event.type == Event.ViewActivateEvent:
+            # Make sure the LayerPass is created
+            self.getLayerPass()
 
-        if self._cancel:
-            return
+            if not self._layerview_composite_shader:
+                self._layerview_composite_shader = OpenGL.getInstance().createShaderProgram(os.path.join(PluginRegistry.getInstance().getPluginPath("LayerView"), "layerview_composite.shader"))
 
-        Job.yieldThread()
-        jump_mesh = layer_data.getLayer(self._layer_number).createJumps()
-        if not jump_mesh or jump_mesh.getVertices() is None:
-            jump_mesh = None
+            if not self._composite_pass:
+                self._composite_pass = self.getRenderer().getRenderPass("composite")
 
-        self.setResult({"layers": layer_mesh.build(), "jumps": jump_mesh})
+            self._old_layer_bindings = self._composite_pass.getLayerBindings()
+            layer_bindings = self._old_layer_bindings[:] # make a copy
+            layer_bindings.append("layerview")
+            self._composite_pass.setLayerBindings(layer_bindings)
```

Comment:
Why does it need to go via this construct, rather than via just `self._composite_pass.getLayerBindings().append("layerview")`?